### DISCUSSION TITLE
Functionality to list folders and files in DVC remote via CLI

### DISCUSSION THREAD

**Discussion by ekwska** [treeverse/dvc#10436] [Help] [answered]
Hi all, I have been using DVC for the last few months and it's been great, however I haven't found a method to do the equivalent of `ls` or `tree` on the remote to show all the subfolders/files that are tracked by DVC. For example if I have `dataset.dvc`

```
outs:
- md5: hash.dir
  nfiles: 100
  hash: md5
  path: dataset/
```

And in `dataset/` there is: 

```
dataset/
|_data1
  |_img_1.png
  |_img_2.png
|_data2/
  |_img_1.png
  |_
```

I would like to query the current hashed version in the remote to see what files/folders exist. Perhaps in a newer version of the dataset there is `data3/`. So I could do `dvc remote list dataset/` and it would show a similar file view to above without having to pull anything.

In this situation I want to **avoid** doing `dvc pull`/`dvc checkout` unless necessary because it's an expensive operation (very large dataset of hundreds of GB).  

Does anyone know if this functionality exists? Happy to provide more info if needed! :metal:

**Comment by dberenbaum** [comment, marked as answer]
Have you taken a look at https://dvc.org/doc/command-reference/list?

For example:

```
$ dvc ls -R https://github.com/iterative/dataset-registry use-cases/cats-dogs
data/train/cats/cat.1.jpg
data/train/cats/cat.10.jpg
data/train/cats/cat.100.jpg
data/train/cats/cat.1000.jpg
data/train/cats/cat.101.jpg
data/train/cats/cat.102.jpg
data/train/cats/cat.103.jpg
data/train/cats/cat.104.jpg
data/train/cats/cat.105.jpg
...
```
